lib/fhir_models/bootstrap/model.rb

```diff
@@ -201,16 +202,16 @@ def validate_field(field, value, contained_here, meta, errors)
             errors[field] << "#{meta['path']}: expected Reference, found #{klassname}"
           end
         # if the data type is a particular resource or complex type or BackBone element within this resource
-        elsif FHIR::RESOURCES.include?(datatype) || FHIR::TYPES.include?(datatype) || v.class.name.start_with?(self.class.name)
+        elsif module_version::RESOURCES.include?(datatype) || module_version::TYPES.include?(datatype) || v.class.name.start_with?(self.class.name)
           if datatype == klassname
             validation = v.validate(contained_here)
             errors[field] << validation unless validation.empty?
           else
             errors[field] << "#{meta['path']}: incorrect type. Found #{klassname} expected #{datatype}"
           end
         # if the data type is a primitive, test the regular expression (if any)
-        elsif FHIR::PRIMITIVES.include?(datatype)
-          primitive_meta = FHIR::PRIMITIVES[datatype]
+        elsif module_version::PRIMITIVES.include?(datatype)
+          primitive_meta = module_version::PRIMITIVES[datatype]
```

Comment:
Currently `primitive?` is not versioned and just under the FHIR module.  I think it makes sense to make it a versioned method in case primitives change from version to version.

So, it doesn't currently need an update, but we can change primitive to be versioned, and then it would need to be.

Reply:
At a glance, I don't think `primitive?` supports all of the types in `PRIMITIVES`.

Reply:
You're right, it was missing canonical, uuid, and url.  each case has been added now.  I'm still wondering if maybe we want this to be a versioned method because that might change from version to version?

Reply:
I don't think it's worth spending time on now. If it's an issue it can be addressed later.